engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/types/EspTypeUtils.scala

```diff
@@ -17,18 +17,20 @@ object EspTypeUtils {
 
   def clazzDefinition(clazz: Class[_])
                      (implicit settings: ClassExtractionSettings): ClazzDefinition =
-    ClazzDefinition(Typed(clazz), extractPublicMethodAndFields(clazz))
+    ClazzDefinition(Typed(clazz),
+      extractPublicMethodsAndFields(clazz, staticMethodsAndFields = false),
+      extractPublicMethodsAndFields(clazz, staticMethodsAndFields = true))
 
-  private def extractPublicMethodAndFields(clazz: Class[_])
+  private def extractPublicMethodsAndFields(clazz: Class[_], staticMethodsAndFields: Boolean)
                                           (implicit settings: ClassExtractionSettings): Map[String, List[MethodInfo]] = {
     val membersPredicate = settings.visibleMembersPredicate(clazz)
-    val methods = extractPublicMethods(clazz, membersPredicate)
-    val fields = extractPublicFields(clazz, membersPredicate).mapValuesNow(List(_))
+    val methods = if(staticMethodsAndFields) extractPublicMethods(clazz, membersPredicate)._2 else extractPublicMethods(clazz, membersPredicate)._1
+    val fields = if(staticMethodsAndFields) extractPublicFields(clazz, membersPredicate)._2.mapValuesNow(List(_)) else extractPublicMethods(clazz, membersPredicate)._1
     methods ++ fields
   }
 
   private def extractPublicMethods(clazz: Class[_], membersPredicate: VisibleMembersPredicate)
-                                  (implicit settings: ClassExtractionSettings): Map[String, List[MethodInfo]] = {
+                                  (implicit settings: ClassExtractionSettings): Tuple2[Map[String, List[MethodInfo]], Map[String, List[MethodInfo]]] = {
```

Comment:
I think it would be easier/nicer if you just pass isStatic flag here?

Reply:
It's much better this way, thanks